syntax = "proto3";
package rpc_server;

// 保存公共数据类型，枚举等

// 服务类型：客户端与服务器通信的服务类型
// 考虑到有些服务需要服务器主动发送消息，以及格式化，所以为所有服务添加响应类型
enum ServiceType
{
    // REQ_ 请求
    // RES_ 响应
    REQ_REGISTER = 0;   // 注册请求
    RES_REGISTER = 1;   // 注册响应
    REQ_LOGIN = 2;      // 登录请求
    RES_LOGIN = 3;      // 登录响应
    REQ_LOGOUT = 4;     // 登出请求
    RES_LOGOUT = 5;     // 登出响应
    REQ_CHANGE_PASSWORD = 6;    // 修改密码请求
    RES_CHANGE_PASSWORD = 7;    // 修改密码响应
    REQ_HEARTBEAT = 8;  // 心跳请求
    RES_HEARTBEAT = 9;  // 心跳响应
    REQ_FILE_TRANSMISSION_READY = 10;    // 文件传输准备请求（获取文件服务器直连）
    RES_FILE_TRANSMISSION_READY = 11;    // 文件传输准备响应
    REQ_FILE_UPLOAD = 12;    // 文件上传请求
    RES_FILE_UPLOAD = 13;    // 文件上传响应
    REQ_FILE_DOWNLOAD = 14;    // 文件下载请求
    RES_FILE_DOWNLOAD = 15;    // 文件下载响应
    REQ_FILE_DELETE = 16;    // 文件删除请求
    RES_FILE_DELETE = 17;    // 文件删除响应
    REQ_FILE_LIST = 18;    // 获取文件列表请求
    RES_FILE_LIST = 19;    // 获取文件列表响应
}

// 日志分级
enum LogLevel
{
    DEBUG = 0;  // debug
    INFO = 1;   // 正常信息
    WARN = 2;   // 警告
    ERROR_LEVEL = 3;    // 异常，ERROR 是宏定义，所以使用 ERROR_LEVEL代替
    CRITICAL = 4;  // 致命错误
    TRACE = 5;   // 跟踪
};

// 日志分类
enum LogCategory
{
    STARTUP_SHUTDOWN = 0;       // 启动和关闭
    APPLICATION_ACTIVITY = 1;   // 应用程序活动（请求、响应、调试）
    CONNECTION_POOL = 2;        // 连接池
    SYSTEM_MONITORING = 3;      // 系统监控（性能监控、定时任务）
    HEARTBEAT = 4;              // 心跳
    SECURITY = 5;               // 安全
    CONFIGURATION_CHANGES = 6;  // 配置变更
    DATABASE_OPERATIONS = 7;    // 数据库操作
    USER_ACTIVITY = 8;          // 用户活动
    NETWORK = 9;                // 网络
};